ChangeStaticFieldToMethod has issue when changing nested code

I'll submit a test for this, but we uncovered the following situation when using the apache commons-collection 3->4 upgrade recipe.

We expect that `ChangeStaticFieldToMethod` will be able to change this:

```
import java.util.Collections;
class Foo {
    final Object o = java.util.Arrays(Collections.EMPTY_LIST);
}
```

into

```
import java.util.Lists;
class Foo {
    final Object o = java.util.Arrays(Lists.of());
}
```

However, when we use the recipe (via that composite recipe), we get a ClassCastException [here](https://github.com/openrewrite/rewrite/blob/2b3291e948cba4e4b99124b4c94c402a8bd11f6a/rewrite-java/src/main/java/org/openrewrite/java/ChangeStaticFieldToMethod.java#L119):

```
Caused by: org.openrewrite.internal.RecipeRunException: java.lang.ClassCastException: class org.openrewrite.java.tree.J$NewArray cannot be cast to class org.openrewrite.java.tree.J$Block (org.openrewrite.java.tree.J$NewArray and org.openrewrite.java.tree.J$Block are in unnamed module of loader 'app')
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:287)
	at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:317)
[snip]
Caused by: java.lang.ClassCastException: class org.openrewrite.java.tree.J$NewArray cannot be cast to class org.openrewrite.java.tree.J$Block (org.openrewrite.java.tree.J$NewArray and org.openrewrite.java.tree.J$Block are in unnamed module of loader 'app')
	at org.openrewrite.java.ChangeStaticFieldToMethod$1.useNewMethod(ChangeStaticFieldToMethod.java:119)
	at org.openrewrite.java.ChangeStaticFieldToMethod$1.visitFieldAccess(ChangeStaticFieldToMethod.java:95)
	at org.openrewrite.java.ChangeStaticFieldToMethod$1.visitFieldAccess(ChangeStaticFieldToMethod.java:80)
	at org.openrewrite.java.tree.J$FieldAccess.acceptJava(J.java:1943)
	at org.openrewrite.java.tree.J.accept(J.java:59)
	at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:245)
```

## What version of OpenRewrite are you using?

rewrite's main branch

## How are you running OpenRewrite?

IDE - running new test in `ChangeStaticFieldToMethodTest`.  I'll push a draft PR soon.

--- a/rewrite-java-test/src/test/java/org/openrewrite/java/ChangeStaticFieldToMethodTest.java
+++ b/rewrite-java-test/src/test/java/org/openrewrite/java/ChangeStaticFieldToMethodTest.java
@@ -166,6 +166,34 @@ class ChangeStaticFieldToMethodTest implements RewriteTest {
     }
 
     @Test
+    @Issue("https://github.com/openrewrite/rewrite/issues/4382")
+    void migratesNestedFieldInitializer() {
+        rewriteRun(
+          java(acmeLists),
+          java(
+            """
+              import java.util.Collections;
+
+              class Foo {
+                  void bar() {
+                      final Object collection = java.util.Arrays.asList(Collections.EMPTY_LIST);
+                  }
+              }
+              """,
+            """
+              import com.acme.Lists;
+
+              class Foo {
+                  void bar() {
+                      final Object collection = java.util.Arrays.asList(Lists.of());
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
     void ignoresUnrelatedFields() {
         rewriteRun(
           java(

--- a/rewrite-java-test/src/test/java/org/openrewrite/java/ChangeStaticFieldToMethodTest.java
+++ b/rewrite-java-test/src/test/java/org/openrewrite/java/ChangeStaticFieldToMethodTest.java
@@ -265,7 +265,7 @@ class ChangeStaticFieldToMethodTest implements RewriteTest {
           java(
             """
               package com.example;
-                            
+
               class Test {
                   public static Object EXAMPLE = null;
               }
@@ -304,7 +304,7 @@ class ChangeStaticFieldToMethodTest implements RewriteTest {
                   private HttpResponseStatus(int code) {
                       this.code = code;
                   }
-                  
+
                   String codeAsText() {
                       return String.valueOf(code);
                   }

--- a/rewrite-java-test/src/test/java/org/openrewrite/java/marker/JavaSourceSetTest.java
+++ b/rewrite-java-test/src/test/java/org/openrewrite/java/marker/JavaSourceSetTest.java
@@ -48,7 +48,7 @@ class JavaSourceSetTest {
           .filter(o -> o.getFullyQualifiedName().startsWith("org.apache.hadoop.hbase.CacheEvictionStats"))
           .findAny();
         assertThat(shaded).isPresent();
-        assertThat(jss.getTypeToGav().get(shaded.get())).isEqualTo("org.apache.hbase:hbase-shaded-client:2.4.11");
+        assertThat(jss.getGavToTypes().get("org.apache.hbase:hbase-shaded-client:2.4.11")).contains(shaded.get());
     }
 
     @Test

--- a/rewrite-java/src/main/java/org/openrewrite/java/ChangeStaticFieldToMethod.java
+++ b/rewrite-java/src/main/java/org/openrewrite/java/ChangeStaticFieldToMethod.java
@@ -116,10 +116,17 @@ public class ChangeStaticFieldToMethod extends Recipe {
 
                 Cursor statementCursor = getCursor().dropParentUntil(Statement.class::isInstance);
                 Statement statement = statementCursor.getValue();
-                J.Block block = makeNewMethod(newClass).apply(statementCursor, statement.getCoordinates().replace());
-                J.MethodInvocation method = block.getStatements().get(0).withPrefix(tree.getPrefix());
+                J applied = makeNewMethod(newClass).apply(statementCursor, statement.getCoordinates().replace());
 
-                if (method.getMethodType() == null) {
+                J.MethodInvocation method = null;
+                if (applied instanceof J.Block) {
+                    J.Block block = (J.Block) applied;
+                    method = block.getStatements().get(0).withPrefix(tree.getPrefix());
+                } else if (applied instanceof J.NewArray) {
+                    J.NewArray newArray = (J.NewArray) applied;
+                    method = (J.MethodInvocation) newArray.getInitializer().get(0);
+                }
+                if (method == null || method.getMethodType() == null) {
                     throw new IllegalArgumentException("Error while changing a static field to a method. The generated template using a the new class ["
                                                        + newClass + "] and the method [" + newMethodName + "] resulted in a null method type.");
                 }

